homeassistant/components/onvif/parsers.py

```diff
@@ -214,6 +214,35 @@ async def async_parse_cell_motion_detector(uid: str, msg) -> Event:
         return None
 
 
+@PARSERS.register("tns1:RuleEngine/LineDetector/Crossed")
+# pylint: disable=protected-access
+async def async_parse_line_crossed(uid: str, msg) -> Event:
+    """Handle parsing event message.
+
+    Topic: tns1:RuleEngine/LineDetector/Crossed
+    """
+    try:
+        video_source = ""
+        video_analytics = ""
+        rule = ""
+        for source in msg.Message._value_1.Source.SimpleItem:
+            if source.Name == "VideoSourceConfigurationToken":
+                video_source = source.Value
+            if source.Name == "VideoAnalyticsConfigurationToken":
+                video_analytics = source.Value
+            if source.Name == "Rule":
+                rule = source.Value
```

Comment:
Should we complete the loop even if the cases above are true?

Reply:
We can `break` if **all** are true, but as mentioned below - the spec defines only these three and this logic is more of a guard than anything else.

Reply:
Ok.